source/sync/event-library.txt

```diff
@@ -0,0 +1,319 @@
+requirements in heavily-regulated industries, such as healthcare or 
+financial services.
+
+.. example::
+
+   A nurse in a healthcare facility uses an app with the Event Library 
+   enabled. The app presents the nurse with real-time vital signs, 
+   information streamed in from the medical equipment, and the patient's
+   historical treatment information. The app inherently enforces compliance,
+   because its code blocks actions that the nurse should not take based
+   on the data being viewed. The Event Library captures all the information
+   that the nurse sees within the app interface, as well as the actions 
+   that the nurse takes after viewing that information.
+   
+   At some point, this nurse provides treatment to a patient that later 
+   exposes the facility to a malpractice lawsuit. The legal department
+   must review the information that was available to the nurse during the
+   treatment. 
+   
+   The Event Library captures the digital data that the nurse 
+   viewed during treatment, as well as the actions he undertook. By 
+   reviewing this data, the legal team can assess whether the treatment was 
+   reasonable. Without this information, the legal department can't know
+   and prove whether the nurse's actions were reasonable.
+
+How it Works
+~~~~~~~~~~~~
+
+The Event Library opens a separate "event" realm on the user's device. This 
+realm has access to any user realm that the developer chooses to monitor
+with the Event Library.
+
+When developers implement the Event Library, they designate the types of 
+events they want to record, as well as any custom metadata they want to 
+append to the event recordings. When the client application runs, it records
+the designated user interactions to the "event" realm as read events, 
+write events, or custom events. 
+
+While the client device has a network connection, {+sync+} synchronizes 
+this event realm data to an ``AuditEvent`` collection in the linked Atlas 
+data source.
+
+.. seealso::
+
+   For information on how to implement the Event Library in a client 
+   application, see: :ref:`Event Library - Swift SDK <swift-event-library>`.
+
+Access Event Library Data
+-------------------------
+
+The Event Library records event data in collection called ``AuditEvent`` 
+in your linked Atlas database. You must create an ``AuditEvent`` collection
+before you use the Event Library to record client application events.
+
+.. seealso::
+
+   For information on how to create a collection, see: :ref:`Create a 
+   Collection - Atlas UI <atlas-ui-create-a-collection>`.
+
+Event Library Schema
+~~~~~~~~~~~~~~~~~~~~
+
+Your ``AuditEvent`` collection must have a schema containing the following
+fields:
+
+- ``_id``: ``ObjectId``
+- ``activity``: ``String``
+- ``event``: ``String?``
+- ``data``: ``String?``
+- ``timestamp``: ``Date``
+
+Additionally, the schema must contain an optional string field (``String?``)
```

Comment:
In JSON schema all fields are optional by default so this would be `{ "<Metadata Key>": { "bsonType": "string" } }`

Reply:
Great point. I've updated this based on the schema object I got from trying out this feature, and am adding this example to the page for folks who might have custom metadata. Thanks!